inc/Base/Cron.php

```diff
@@ -113,150 +113,223 @@ public function smaily_sync_contacts() {
+			// Required for legacy API so that all fields are always updated.
+			foreach ( $cart_sync_values as $key ) {
+				for ( $i = 1; $i < 11; $i++ ) {
+					$addresses[ $key . '_' . $i ] = '';
 				}
-				// Gather products data if user has selected at least one of additional product field to sync.
-				if ( ! empty( array_intersect( $cart_sync_values, $results['cart_options'] ) ) ) {
-					$products_data = [];
-					foreach ( $cart_data as $cart_item ) {
-						// Single product detais array.
-						$product = [];
-						$details = wc_get_product( $cart_item['product_id'] ); // Get product by ID.
-						// Get product details if selected from user settings.
-						if ( in_array( 'product_name', $results['cart_options'], true ) ) {
-							$product['name'] = htmlspecialchars( $details->get_name() );
-						}
-						if ( in_array( 'product_description', $results['cart_options'], true ) ) {
-							$product['description'] = htmlspecialchars( $details->get_description() );
-						}
-						if ( in_array( 'product_sku', $results['cart_options'], true ) ) {
-							$product['sku'] = htmlspecialchars( $details->get_sku() );
-						}
-						if ( in_array( 'product_description_short', $results['cart_options'], true ) ) {
-							$product['description_short'] = htmlspecialchars( $details->get_short_description() );
-						}
-						if ( in_array( 'product_quantity', $results['cart_options'], true ) ) {
-							$product['quantity'] = htmlspecialchars( $cart_item['quantity'] );
-						}
-						if ( in_array( 'product_subtotal', $results['cart_options'], true ) ) {
-							$product['subtotal'] = htmlspecialchars( $cart_item['line_subtotal'] );
+			}
+			$selected_fields = array_intersect( $cart_sync_values, $results['cart_options'] );
+			// Gather products data if user has selected at least one of additional product field to sync.
+			if ( ! empty( $selected_fields ) ) {
+				$products_data = [];
+				foreach ( $cart_data as $cart_item ) {
+					// Single product detais array.
+					$product = [];
+					// Get product details if selected from user settings.
+					$details = wc_get_product( $cart_item['product_id'] );
+					foreach ( $selected_fields as $selected_field ) {
+						switch ( $selected_field ) {
+							case 'product_name':
+								$product['product_name'] = $details->get_name();
+								break;
+							case 'product_description':
+								$product['product_description'] = $details->get_description();
+								break;
+							case 'product_sku':
+								$product['product_sku'] = $details->get_sku();
+								break;
+							case 'product_quantity':
+								$product['product_quantity'] = $cart_item['quantity'];
+								break;
+							case 'product_price':
+								$product['product_price'] = $this->get_sale_price( $details );
+								break;
+							case 'product_base_price':
+								$product['product_base_price'] = $this->get_base_price( $details );
+								break;
+							default:
```

Comment:
Currently, this default field is unnecessary because there are no more options available. Should there be some kind of error logging when reaching the default case or should I just omit this?

Reply:
You can just omit it.

Reply:
Done https://github.com/sendsmaily/smaily-woocommerce-plugin/pull/37/commits/c9e01ae4005c49a915ac6aaf9450a432e909747b